Build failing due to specifying template argument in ctor

I could not build the project with the following strange errors.
```
> ninja
[3/34] Performing configure step for 'elfio'
Skipping configure step.
[4/34] Performing build step for 'elfio'
Skipping build step.
[5/34] Performing install step for 'elfio'
Skipping install step.
[7/34] Building CXX object CMakeFiles/mipt-mips.dir/export/standalone/main.cpp.o
FAILED: CMakeFiles/mipt-mips.dir/export/standalone/main.cpp.o
/usr/bin/c++ -DKERNEL_IMAGES=\"/home/andrew/mipt-mips/simulator/../kernels/\" -DTEST_PATH=\"/home/andrew/mipt-
w/mipt-mips/simulator/elfio/src/elfio -isystem /home/andrew/mipt-mips/simulator/../external -isystem /home/and
t -Wvla -Werror -O3 -DNDEBUG -O3 -march=native -std=c++20 -MD -MT CMakeFiles/mipt-mips.dir/export/standalone/m
es/mipt-mips.dir/export/standalone/main.cpp.o -c /home/andrew/mipt-mips/simulator/export/standalone/main.cpp
In file included from /home/andrew/mipt-mips/simulator/infra/ports/module.h:11,
                 from /home/andrew/mipt-mips/simulator/simulator.h:12,
                 from /home/andrew/mipt-mips/simulator/kernel/kernel.h:13,
                 from /home/andrew/mipt-mips/simulator/export/standalone/main.cpp:10:
/home/andrew/mipt-mips/simulator/infra/ports/ports.h:129:19: error: expected unqualified-id before ‘const’
  129 |     WritePort<T>( const std::shared_ptr<PortMap>& port_map, const std::string& key, uint32 bandwidth)
      |                   ^~~~~
/home/andrew/mipt-mips/simulator/infra/ports/ports.h:129:18: error: expected ‘)’ before ‘const’
  129 |     WritePort<T>( const std::shared_ptr<PortMap>& port_map, const std::string& key, uint32 bandwidth)
      |                 ~^~~~~~
      |                  )
/home/andrew/mipt-mips/simulator/infra/ports/ports.h:156:18: error: expected unqualified-id before ‘const’
  156 |     ReadPort<T>( const std::shared_ptr<PortMap>& port_map, const std::string& key, Latency latency)
      |                  ^~~~~
/home/andrew/mipt-mips/simulator/infra/ports/ports.h:156:17: error: expected ‘)’ before ‘const’
  156 |     ReadPort<T>( const std::shared_ptr<PortMap>& port_map, const std::string& key, Latency latency)
      |                ~^~~~~~
      |                 )

```
After trying to remove imho unnecessary `<T>`s everything worked like a charm. Apparently, no one has ever encountered such a problem (judging by the absence of related issues). I am using gcc 11, boost 1.75, latest cmake, latest kernel. Can provide further info if needed. 

I wonder if you would be willing to fix this or accept a PR?

--- a/simulator/infra/ports/ports.h
+++ b/simulator/infra/ports/ports.h
@@ -126,7 +126,7 @@ template<class T> class ReadPort;
 template<class T> class WritePort : public BasicWritePort
 {
 public:
-    WritePort<T>( const std::shared_ptr<PortMap>& port_map, const std::string& key, uint32 bandwidth)
+    WritePort( const std::shared_ptr<PortMap>& port_map, const std::string& key, uint32 bandwidth)
         : BasicWritePort( port_map, key, bandwidth)
     { }
 
@@ -153,7 +153,7 @@ private:
 template<class T> class ReadPort : public BasicReadPort
 {
 public:
-    ReadPort<T>( const std::shared_ptr<PortMap>& port_map, const std::string& key, Latency latency)
+    ReadPort( const std::shared_ptr<PortMap>& port_map, const std::string& key, Latency latency)
         : BasicReadPort( port_map, key, latency)
     { }
 

